Code_Exercises/Exercise_18_Local_Memory_Tiling/solution.cpp

```diff
@@ -93,6 +93,12 @@ TEST_CASE("image_convolution_tiled", "local_memory_tiling_solution") {
                     auto localId = item.get_local_id();
                     auto globalGroupOffset = groupId * localRange;
 
+                    // Explanation:
+                    // We need to read pixels from the neighbouring work-groups 
```

Comment:
Good description. I might reword this to say something more like:

Each work group will need to read a tile of size `(localRange[0] + halo * 2, localRange[1] + halo * 2)`
in order to write a tile of size `(localRange[0], localRange[1])`. Since the size of the tile we need to
read is larger than the workgroup size (localRange), we must do multiple loads per work item.
The iterations of the for loop work as follows:

```
          <- localRange[0] + halo *2 ->
          +------------------------------+  ^
          |+-----------------++---------+|  |
        ^ ||<-localRange[0]->||         ||  |
        | ||                 ||         ||  |
    local ||   iteration 1   ||  it 2   ||  |
 Range[1] ||     load        ||  load   ||
        | ||                 ||         ||  localRange[1] +
        | ||                 ||         ||  halo * 2
        V ||                 ||         ||
          |+-----------------++---------+|  |
          |+-----------------++---------+|  |
          ||    it  3 load   ||it 4 load||  |
          |+-----------------++---------+|  |
          +------------------------------+  V
```

Reply:
Yeah that's a good way of putting it and the graphic definitely helps. I'm happy to use this description

Reply:
Great thanks